packages/test-utils/src/wrapper.js

```diff
@@ -711,7 +711,17 @@ export default class Wrapper implements BaseWrapper {
 
       // $FlowIgnore : Problem with possibly null this.vm
       this.vm.$forceUpdate()
-      return nextTick()
+      return new Promise(resolve => {
+        nextTick().then(() => {
+          const isUpdated = Object.keys(data).some(key => {
+            return (
+              // $FlowIgnore : Problem with possibly null this.vm
+              this.vm[key] === data[key] || this.vm.$attrs[key] === data[key]
+            )
+          })
+          return !isUpdated ? this.setProps(data).then(resolve()) : resolve()
```

Comment:
Should this be
`.then(() => resolve())`
instead of
`.then(resolve())`

?

Reply:
Additionally, the signature on setProps claims to potentially return void
```typescript
interface BaseWrapper {
	setProps (props: object): Promise<void> | void
}
```

so I'm getting warnings on
```typescript
? this.setProps(data).then(`
```
```
property 'then' does not exist on type 'void | Promise<void>'.
  Property 'then' does not exist on type 'void'
```

maybe I'm missing something, I'm just trying to backport the fix for my project

<details>
<summary>Janky backport function if anyone cares</summary>

```typescript
function setPropsForSure(
	wrapper: Wrapper<CombinedVueInstance<any, any, any, any, any>>,
	data: any,
) {
	return new Promise(resolve => {
		wrapper.vm.nextTick().then(() => {
			const isUpdated = Object.keys(data).every(key => {
				return (
					// $FlowIgnore : Problem with possibly null this.vm
					wrapper.vm[key] === data[key] || wrapper.vm.$attrs[key] === data[key]
				);
			});
			if (isUpdated) {
				return resolve();
			}
			let result = wrapper.setProps(data);
			if (result) {
				result.then(() => resolve());
			} else {
				resolve();
			}
		});
	});
}
```

</details>